### ISSUE TITLE
A question regarding 3D convolutions

### ISSUE THREAD

**Issue by johanneskvam** [NVIDIA/MinkowskiEngine#270]
Hi, 

I've been trying to calculate the effective receptive field of a Minkowski UNet architecture on point clouds, and I've had some strange results, indicating that the 3D convolutions are only operating in a single plane. This led me to play with some simple toy examples, and I've encountered a suprising behaviour. 

For a simple network composed of a single MinkowskiConvolution of dimension 3, it appears that strides are not operating as I expected. The strides appear to be working as expected in the x and y direction, but not in the z direction. Below is the example code I have been playing around with and two visualizations of the output cube after the strided convolution. 

    device = torch.device('cuda' if torch.cuda.is_available() else 'cpu')
    # Model, single 3D convolution with strides in x,y and z.
    model = nn.Sequential(
        ME.MinkowskiConvolution(in_channels=3, out_channels=3, kernel_size=5, strides=(3,3,3), dimension=3)).to(device)
    
    voxel_size = 0.08

    # Input cloud (just a cube)
    xyz = np.stack(np.meshgrid(np.arange(-2,2,voxel_size), np.arange(-2,2, voxel_size), np.arange(-2,2, voxel_size)))
    xyz = np.swapaxes(xyz.reshape(xyz.shape[0], np.product(xyz.shape[1:])), 0,1)
    
    # RGB features, all white
    rgb = np.ones_like(xyz) * 255
    
    # Quantize
    coords = np.floor(xyz / voxel_size)
    inds = ME.utils.sparse_quantize(coords,return_index=True)
    coords = coords[inds]

    coords = torch.tensor(np.hstack([coords, np.zeros((len(coords), 1))]), dtype=torch.int32)
    feats = torch.tensor(rgb[inds], dtype=torch.float32)
    xyz_d = xyz[inds]
    
    # Sparse input tensor
    X = ME.SparseTensor(torch.ones_like(feats), coords=coords).to(device)
    
    # Run through network
    o = model(X)
    # Export cloud for visualization
    export_cloud(o.coords[:,:3].numpy())

Visualization of cloud in xy-plane
![image](https://user-images.githubusercontent.com/7849437/101169634-88258f80-363d-11eb-8c1b-b5d65745adc6.png)
Visualization of cloud in xz-plane
![image](https://user-images.githubusercontent.com/7849437/101169696-9f647d00-363d-11eb-95ad-654994761b70.png)

This behaviour is something I've also encountered when calculating the receptive field, as it appears that for a stack of convolutions the receptive field is essentially only a 2D plane. Below is an image, where the seed point is colored in red. 

![image](https://user-images.githubusercontent.com/7849437/101171335-e3588180-363f-11eb-88b7-58cd403eedbf.png)

I'm probably missing something obvious here, but I've been going through the docs and cannot find any reason for this. I hope you can shed some light on whats going on.

**Comment by johanneskvam** [comment]
I figured out that an upgrade to MinkowskiEngine 0.4.3, changed notation in coordinates to batch first. Therefore, one of the convolutions were likely done across the batch index.
